demo/image-segmentation/pytorch_loader.py

```diff
@@ -282,9 +282,19 @@ def __getitems__(self, indices):
 
         res = []
         for i in range(len(images_in_bytes)):
+            try:
```

Comment:
It's ok to catch and skip for the purpose of completing the training successfully but it also will make the # data samples to be smaller than what's supposed to be. Is that desired?

Reply:
Also, I thought we ran into the same problem with the native way of loading datasets so that's going to be line 182. Do we need that to be modified?

Reply:
Finally, line 241 and 251 also does the data loading, do we need them modified too?

Reply:
1. I logged the names of the files that were causing the error. There weren't more than 3 to 4 problematic files. I believe that doesn't affect/minimally affects the metrics we are trying to collect (per-step training time) or the goal of this demo which is to show the benefits of dataflux (correct me if I'm wrong).

2, 3. The errors I ran into were with loading numpy arrays from raw bytes like here. Nevertheless I think it's good to wrap all calls to np.load in try-except.

Reply:
> 2, 3. The errors I ran into were with loading numpy arrays from raw bytes like here.

Yes this is expected because we defined the "__get_items__" method so PyTorch DataLoader will invoke this method. However, it will be good for the "__getitem__" method to incorporate this change. Similar for the native data loading.

Reply:
Working on those changes! Although I'm not sure what to return when an exception occurs. Probably a random numpy array whose dimensions conform to the expectations of the training code. The dataloader currently has no knowledge of that so it's a bit tricky to make that change. Any recommendations?

Reply:
Also, for the purposes of this demo here, I think it suffices to fix only `__getitems__` for now because the training code doesn't reference the files in the dataset by index. Neither does it ask the user to read files from a local directory. I could make changes to those two places for the sake of completeness but I believe it's not necessary as far as this demo is concerned. Happy to spend a little more time figuring out what to return from `__getitem__` when there's an error if you feel strongly that it makes sense to do it now.

Reply:
Probably should just raise the error instead.

`training code doesn't reference the files in the dataset by index` --> It still does. It just references a range of indices instead, when the `__getitems__` method is defined.

I would think that we should just keep all places consistent, for the sake of code quality. It would be confusing to see different data loading mechanisms, especially when the demo is instructing on how to **switch** from one dataset to the one with Dataflux.

Reply:
There are some pretty extensive conversations on how to handle exceptions like this if you search for it, e.g. https://www.google.com/search?q=pytorch+data+loader+exceptions&oq=pytorch+d&gs_lcrp=EgZjaHJvbWUqCAgAEEUYJxg7MggIABBFGCcYOzIGCAEQRRg5MgYIAhBFGDwyBggDEEUYPDIGCAQQRRg8MgYIBRBFGEAyBggGEEUYQDIGCAcQRRhAqAIAsAIA&sourceid=chrome&ie=UTF-8. Is the approach here consistent with what those conversations are generally recommending?

Reply:
Followed https://github.com/pytorch/pytorch/issues/1137#issuecomment-618286571. Made changes to all `__getitem__` and `__getitems__` methods.